src/wrappers/pubsub.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @fileoverview Handlers for Pubsub instrumentation
+ */
+
+const shimmer = require('shimmer');
+const {
+    tracer,
+    tryRequire,
+    eventInterface,
+    utils,
+} = require('epsagon');
+const traceContext = require('../trace_context.js');
+
+const subscriber = tryRequire('@google-cloud/pubsub/build/src/subscriber');
+
+/**
+ * Handle subscriber event emitter of eventName='message'
+ * @param {Message} message received message.
+ * @param {*} originalHandler listener callback function.
+ * @param {*} requestFunctionThis request arguments.
+ */
+function pubSubSubscriberMiddleware(message, originalHandler, requestFunctionThis) {
+    let originalHandlerSyncErr;
+    try {
+        // Initialize tracer and evnets.
+        tracer.restart();
+        const { slsEvent: pubSubEvent, startTime: pubSubStartTime } =
+        eventInterface.initializeEvent(
+            'pubsub',
+            requestFunctionThis.projectId,
+            'messagePullingListener',
+            'trigger'
+        );
+        tracer.addEvent(pubSubEvent);
+
+        // Getting message data.
+        let runnerMetadata = { messageId: message.id };
+        const messageData = (message.data && JSON.parse(`${message.data}`));
+        if (messageData && typeof messageData === 'object') {
+            runnerMetadata = Object.assign(runnerMetadata, messageData);
+        }
+        const { label, setError } = tracer;
+        // eslint-disable-next-line no-param-reassign
+        message.epsagon = {
+            label,
+            setError,
+        };
+
+        // Finalize pubsub event.
+        eventInterface.finalizeEvent(pubSubEvent, pubSubStartTime, null, runnerMetadata);
```

Comment:
move before line 42 (that's where you are done with dealing with the trigger and start messing with the runner. It also might be nicer to call tracer.addEvent(pubSubEvent) below this line.

Reply:
agree.